lib/msf/core/post/file.rb

```diff
@@ -501,19 +501,22 @@ def rm_rf(*remote_dirs)
   alias :dir_rm :rm_rf
 
   #
-  # Rename a remote file.
+  # Renames a remote file and returns true on success and false
+  # on failure
   #
   # @param old_file [String] Remote file name to move
   # @param new_file [String] The new name for the remote file
   def rename_file(old_file, new_file)
+    verification_token = Rex::Text.rand_text_alphanumeric(8)
     if session.type == "meterpreter"
       return (session.fs.file.mv(old_file, new_file).result == 0)
+    elsif session.type == 'powershell'
+      !!(cmd_exec("Rename-Item -Path \"#{old_file}\" -NewName \"#{new_file}\"; if($?){echo #{verification_token}}") =~ /#{verification_token}/)
```

Comment:
This doesn't work to move the file into a new directory.

```
PS C:\Users\smcintyre\Desktop> Rename-Item -Path .\disassembled.txt -NewName C:\Users\smcintyre\
Rename-Item : Cannot rename the specified target, because it represents a path or device name.
At line:1 char:1
+ Rename-Item -Path .\disassembled.txt -NewName C:\Users\smcintyre\
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Rename-Item], PSArgumentException
    + FullyQualifiedErrorId : Argument,Microsoft.PowerShell.Commands.RenameItemCommand

PS C:\Users\smcintyre\Desktop>
```

We'll need a different solution that can move the file into another directory. It should then handle cases where the destination is a directory like the others do (by moving the file into that directory).

Reply:
I guess using `Rename-Item` cmdlet is not a good idea as it just renames the item in the same directory, `Move-Item` should do I guess.

Reply:
fixed :+1: